kernel/private/classes/exceptions/database/exception.php

```diff
@@ -15,7 +15,31 @@
  * @version //autogentag//
  * @package kernel
  */
-class eZDBException extends ezcBaseException
+class eZDBException extends Exception
```

Comment:
break

Reply:
What's here now is essentially a copy of ezcBaseException, but with support for the error code / number. Thus, the intention is that there is no BC break at all.

Reply:
It's a break if someone typehinted against `ezcBaseExtension`.

Can you not implement a custom base exception extending from `ezcBaseExtension` with added features?

Reply:
It's unlikely somebody would typehint against 'ezcBaseException'. It's much more likely that somebody would typehint against 'eZDBException'

Reply:
Sure, but that doesn't change the fact that it's still a BC break which IMO should not be made lightly.

Reply:
OK, I've rewritten it so that we continue to extend ezcBaseException but just set the $message and $code properties, since the ezcBaseException constructor removed $code support.